srcjs/_start.js

```diff
@@ -3,6 +3,17 @@
 
   var exports = window.Shiny = window.Shiny || {};
 
+  (function(history){
+    var pushState = history.pushState;
+    history.pushState = function(state) {
+        if (typeof history.onpushstate == "function") {
```

Comment:
Also, does this mean you for the "monkey-path" version, instead of the "hack" version (given that the latter doesn't need this function at all)?